sdk/python/kfp/dsl/_pipeline_param.py

```diff
@@ -42,7 +53,7 @@ def match_serialized_pipelineparam(payload: str):
   for match in matches:
       pattern = '{{pipelineparam:op=%s;name=%s}}' % (match[0], match[1])
       param_tuples.append(PipelineParamTuple(
-                          name=sanitize_k8s_name(match[1]), 
+                          name=sanitize_k8s_name(match[1], True),
                           op=sanitize_k8s_name(match[0]), 
```

Comment:
Are you sure the op name cannot have underscores and capital letters?

Reply:
It supports capital letters, but not underscores. The error I hit: `invalid spec: spec.templates[2].name: 'dataproc_create_cluster' is invalid: name must consist of alpha-numeric characters or '-', and must start with an alpha-numeric character (e.g. My-name1-2, 123-NAME)`

Reply:
What I intended to do here was, only bypass sanitizing `_` and capital letters when users have the need. I don't see that users would access the op names in their code in a Pythonic way so I think it's still okay to sanitize the op names.

WDYT?

Reply:
I just want those limitations to be addressed closer to the source of those limitations - to the compiler.
I also though that maybe the sanitization function can be simpler with only one regex.
Ideally we should move `match_serialized_pipelineparam` to the compiler.
It's OK to do that in another PR.

Reply:
Thanks @Ark-kun . Will do in a following PR.